flink-formats/flink-avro/src/main/java/org/apache/flink/formats/avro/RowDataToAvroConverters.java

```diff
@@ -67,13 +67,28 @@ public static RowDataToAvroConverter createConverter(LogicalType type) {
 		final RowDataToAvroConverter converter;
 		switch (type.getTypeRoot()) {
 		case NULL:
-			converter = (schema, object) -> null;
+			converter = new RowDataToAvroConverter() {
```

Comment:
These changes are because shading doesn't work with lambdas, right? Just putting it here in case someone else is wondering in the future.

It might be worth putting a comment above this section that explains it, otherwise someone else might replace these by lambdas in the future. But then the tests would fail, right?

Reply:
Yes, the test should fail, unless only a converter for DataType that is not used in the test is updated.

I added a comment there.

Reply:
Could we add `@SuppressWarnings("Convert2Lambda")` annotation to the method? This can avoid warning in the IDEA. Besides, it would be better to add `serialVersionUID` to the `RowDataToAvroConverter`.

Reply:
Good point. I added `serialVersionUID`. With it in place there is no need for `@SuppressWarnings("Convert2Lambda")` as the anonymous classes can no longer be converted to a lambda.